cluster-autoscaler/core/utils/pod_schedulable.go

```diff
@@ -71,7 +71,7 @@ func (podMap PodSchedulableMap) Get(pod *apiv1.Pod) (*simulator.PredicateError,
 }
 
 // Set sets scheduling info for given pod in PodSchedulableMap
-func (podMap PodSchedulableMap) Set(pod *apiv1.Pod, err *simulator.PredicateError) {
+func (podMap PodSchedulableMap) Set(pod *apiv1.Pod, err simulator.PredicateError) {
```

Comment:
I'm very worried about this - we rely on nil value of the error to mean that a pod can be scheduled. Given that underlying struct is pointer I'm worried we may write a nil of the pointer type and fail victim to the way go types nil. Either revert the change to interface or make sure we always create the nil of the interface type or use reflect.IsNil when checking for nil or something.

Reply:
Changed.
It is like this now

```golang
// PredicateError represents an error during predicate checking.
type PredicateError *predicateError

type predicateError struct {
	errorType     PredicateErrorType
	predicateName string
	errorMessage  string
	reasons       []string
	// debugInfo contains additional info that predicate doesn't include,
	// but may be useful for debugging (e.g. taints on node blocking scale-up)
	debugInfo func() string
}
```

Reply:
Why typedef for a pointer? I'm not strongly opposed in principle, just curious. It feels a bit redundant and I'm not convinced it helps readability - there is some usefulness to being explicit whether something is passed by value or by reference.

Reply:
I applied Ola's commit which makes us directly use pointer.